### PR TITLE
For #11875 - Prevent mixup of region / locale based search engines

### PR DESCRIPTION
**Pull request by Mugurell**
We have two search engine types:
- one based on MLS reported region,
- one based only on Locale.

There are multiple steps involved in returning the default search engine for
example and though at each step we could verify if a certain operation is
completed we are still exposed to concurrency issues.
Simplest and most effective way to make sure the MLS engines do not mix with
Locale based engines is to use the same type of engines for the entire duration
of the app. At the next cold start we'll verify again which engines to use.

Using the Locale based engines (fallbacks) is expected to only happen once, at
the first run of the application after being installed.

### Pull Request checklist

- [x] **Tests**: No tests. Minor changes.
- [x] **Screenshots**: No tests. Comes to resolve a concurrency issue.
- [x] **Accessibility**: The code in this PR does not include any user facing features.

### After merge
- [ ] **Milestone**: Make sure issues finished by this pull request are added to the [milestone](https://github.com/mozilla-mobile/fenix/milestones) of the version currently in development.

### To download an APK when reviewing a PR:
1. click on Show All Checks,
2. click Details next to "Taskcluster (pull_request)" after it appears and then finishes with a green checkmark,
3. click on the "Fenix - assemble" task, then click "Run Artifacts".
4. the APK links should be on the left side of the screen, named for each CPU architecture

### CODE REVIEW

**Comment by Mugurell** [comment]
This depends on the patch from https://github.com/mozilla-mobile/android-components/issues/7627

File: app/src/main/java/org/mozilla/fenix/components/searchengine/FenixSearchEngineProvider.kt [hunk 1]
```diff
@@ -43,6 +43,13 @@ open class FenixSearchEngineProvider(
         )
     }
 
+    // We have two search engine types: one based on MLS reported region, one based only on Locale.
+    // There are multiple steps involved in returning the default search engine for example.
+    // Simplest and most effective way to make sure the MLS engines do not mix with Locale based engines
+    // is to use the same type of engines for the entire duration of the app's run.
+    // See fenix/issues/11875
+    private val isRegionCachedByLocationService = locationService.hasRegionCached()
```
Comment (boek) [inline app/src/main/java/org/mozilla/fenix/components/searchengine/FenixSearchEngineProvider.kt:51]: ah yes. This is a much better idea 🙌

**Comment by liuche** [comment]
I think we might still need the AC changes to propagate in before we can merge this.

**Comment by Mugurell** [comment]
Closing and reopening the PR for restarting the CI jobs.

**Comment by Mugurell** [comment]
Planned to reopen this to restart CI tasks but seems Github doesn't agree...

**Comment by Mugurell** [comment]
Seems like with the issues GH had this morning at the moment when I force-pushed after rebase this PR was seen as closed.
Aand GH doesn't allow that - https://github.com/isaacs/github/issues/361

I'll open a new PR.

**Comment by Mugurell** [comment]
Opened https://github.com/mozilla-mobile/fenix/pull/12502

**Pull request closed without merging**